wireapp/wire-android
# feat: End call on conversation degraded (WPB-4487)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4487" title="WPB-4487" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4487</a>  [proteus] Show a dialog to explain why a call was ended (conversation degraded status)
  </td></table>
  <br />
 










# What's new in this PR?

When the conversations verification status (MLS or Proteus) degrades from the "Verified" during the call, call should be ended and user should see the dialog informing him why the call was ended

**PS** a lot of changes are just imports updates which caused by changes in kalium, donnow why corresponding PR is absent in AR.

### Testing

#### How to Test

For now kalium part works only for the Proteus.

- verify all your devices
- verify all the devices of some user
- open the conversation with that user and start a call
-  (without ending a call) un-verify some device of that user

**expected result:** call is ended and you see the dialog informing that call was ended because of verification changes

- verify all your devices
- verify all the devices of some user
- open the conversation with that user and start a call
-  (without ending a call) un-verify some of your devices

**expected result:** call is ended and you see the dialog informing that call was ended because of verification changes

### Attachments (Optional)

<img width="280" alt="Screenshot 2023-11-08 at 11 05 17" src="https://github.com/wireapp/wire-android-reloaded/assets/6539347/eff81baa-4c7a-4659-97df-aca9e02586a5">
